src/Nethermind/Nethermind.Synchronization/Peers/SyncPeerPool.cs

```diff
@@ -477,123 +477,6 @@ private bool CanBeUsefulForFastBlocks(long blockNumber)
                    lowestInsertedHeader > 1 && lowestInsertedHeader < blockNumber;
         }
 
-        internal void DropUselessPeers(bool force = false)
-        {
-            if (!force && DateTime.UtcNow - _lastUselessPeersDropTime < TimeSpan.FromSeconds(30))
-                // give some time to monitoring nodes
-                // (monitoring nodes are nodes that are investigating the network but are not synced themselves)
-                return;
-
-            if (_logger.IsTrace) _logger.Trace($"Reviewing {PeerCount} peer usefulness");
-
-            int peersDropped = 0;
-            _lastUselessPeersDropTime = DateTime.UtcNow;
-
-            long ourNumber = _blockTree.BestSuggestedHeader?.Number ?? 0L;
-            UInt256 ourDifficulty = _blockTree.BestSuggestedHeader?.TotalDifficulty ?? UInt256.Zero;
-            foreach (PeerInfo peerInfo in NonStaticPeers)
-            {
-                if (peerInfo.HeadNumber == 0
-                    && peerInfo.IsInitialized
-                    && ourNumber != 0
-                    && peerInfo.PeerClientType != NodeClientType.Nethermind
-                    && peerInfo.PeerClientType != NodeClientType.Trinity)
-                // we know that Nethermind reports 0 HeadNumber when it is in sync (and it can still serve a lot of data to other nodes)
-                {
-                    if (!CanBeUsefulForFastBlocks(peerInfo.HeadNumber))
-                    {
-                        peersDropped++;
-                        peerInfo.SyncPeer.Disconnect(InitiateDisconnectReason.UselessInFastBlocks, "PEER REVIEW / HEAD 0");
-                    }
-                }
-                else if (peerInfo.HeadNumber == 1920000 && _blockTree.NetworkId == BlockchainIds.Mainnet) // mainnet, stuck Geth nodes
-                {
-                    if (!CanBeUsefulForFastBlocks(peerInfo.HeadNumber))
-                    {
-                        peersDropped++;
-                        peerInfo.SyncPeer.Disconnect(InitiateDisconnectReason.UselessInFastBlocks, "PEER REVIEW / 1920000");
-                    }
-                }
-                else if (peerInfo.HeadNumber == 7280022 && _blockTree.NetworkId == BlockchainIds.Mainnet) // mainnet, stuck Geth nodes
-                {
-                    if (!CanBeUsefulForFastBlocks(peerInfo.HeadNumber))
-                    {
-                        peersDropped++;
-                        peerInfo.SyncPeer.Disconnect(InitiateDisconnectReason.UselessInFastBlocks, "PEER REVIEW / 7280022");
-                    }
-                }
-                else if (peerInfo.HeadNumber > ourNumber + 1024L && _betterPeerStrategy.IsLowerThanTerminalTotalDifficulty(peerInfo.TotalDifficulty) && peerInfo.TotalDifficulty < ourDifficulty)
```

Comment:
Shouldnt this stay?

Reply:
I don't see the point, can you show me example scenario where this will be useful?

Reply:
nvm useless